src/postgres/tree/PostgresDatabaseTreeItem.ts

```diff
@@ -120,6 +141,18 @@ export class PostgresDatabaseTreeItem extends AzureParentTreeItem<ISubscriptionC
             } finally {
                 await client.end();
             }
+        } else if (this.parent.connectionString) {
+            const username_connString = nonNullProp(this.parent.connectionString, 'username');
+            const password_connString = nonNullProp(this.parent.connectionString, 'password');
+            const host = nonNullProp(this.parent.connectionString, 'hostName');
+            const clientConfig: ClientConfig = { user: username_connString, password: password_connString, ssl, host, port: 5432, database: this.databaseName };
+            const client: Client = new Client(clientConfig);
+            try {
+                await client.connect();
+                return clientConfig;
+            } finally {
+                await client.end();
+            }
```

Comment:
Is it possible to move this somewhere where it can be shared with the same code above?